download.go

```diff
@@ -1,53 +1,53 @@
 package playwright
 
-import "errors"
-
 type downloadImpl struct {
-	channelOwner
+	page              *pageImpl
+	url               string
+	suggestedFilename string
+	artifact          *artifactImpl
 }
 
 func (d *downloadImpl) String() string {
 	return d.SuggestedFilename()
 }
 
+func (d *downloadImpl) Page() *pageImpl {
+	return d.page
+}
+
 func (d *downloadImpl) URL() string {
-	return d.initializer["url"].(string)
+	return d.url
 }
 
 func (d *downloadImpl) SuggestedFilename() string {
-	return d.initializer["suggestedFilename"].(string)
+	return d.suggestedFilename
 }
 
 func (d *downloadImpl) Delete() error {
-	_, err := d.channel.Send("delete")
+	err := d.artifact.Delete()
```

Comment:
nit: you can return directly everywhere here instead of storing the single erro variable inside a variable.